### ISSUE TITLE
Is it possible to mark a specific key with a color?

### ISSUE THREAD

**Issue by hepiyellow** [garynewby/GLNPianoView#5]
For example:
- To show the middle C key
- or to instruct the user which key to press

**Comment by garynewby** [comment]
I've added a function called highlightKeys which will allow you to do that.

**Comment by hepiyellow** [comment]
Hi, Thanks for the quick enhancement !

1. You made a few breaking changes, some types changed from UInt8 to Int. I made the migration easy, so no problem here.
2. I think the `highlightKeys` function should accept a note numbers in the first argument, instead of note names. Is the `Note` class even exposed in the module? I got unresolved identifier when trying to use it.
3. The highlight doesn't work for black keys (I tried with "Eb5")

**Comment by garynewby** [comment]
I prefer note names for 2. as i think in note names not midi numbers 🙂. I've made the note class public for conversion purposes though, and fixed the note highlighting for black key.

**Comment by hepiyellow** [comment]
Alright, we're on !

Works like a charm.

Two comments though:
1. Just consider that your implementation of note-names (`Note` class) is opinionated. While more generic libraries like `MusicTheorySwifty` allow both "D#5" and "Eb5" Pitches to represent the same note, you allow only "Eb5". That is not a problem if I convert to Pitch to note-number and then convert it to your note-name.
2. The black key highlight is flat (no frame), so if I highlight adjacent white & black keys, it looks weird.
![Screen Shot 2019-11-10 at 9 07 08 AM](https://user-images.githubusercontent.com/6338722/68540408-70c49600-039a-11ea-8bfb-8ddb8810ade2.png)

Thanks again !

**Comment by hepiyellow** [comment]
Found a BUG )-:

- When I call `highlightKeys()` while a note is pressed (down), then triggers an undesirable call to the delegate's `pianoKeyUp`
- This is done in the `reset()` function [here](https://github.com/garynewby/GLNPianoView/blob/243dcbb5e8557236d62936fce28623551f2f8bdb/Source/GLNPianoView.swift#L192)

**Comment by hepiyellow** [comment]
## One more issue
When a highlighted key is pressed it's appearance doesn't change
(As opposed to unhighlighted keys which get a shadow)

## A little feature request
It would be great if I could specify the highlight color of each key separately.

_In my case, I need to highlight the possible keys that are allowed to be played in color A,
and I want to highlight the first note that needs to be played with color B._

So I suggest an API:
```
highlightKeys([(keyNumber: Int, color: UIColor)])
```
* I think the `play: Bool` argument is redundant. The consumer can do it themselves.
```

**Comment by garynewby** [comment]
I'll look at, when I have time - some of your highlighting issues can be addressed by using a highlight colour with alpha < 1, so you see the image below.

**Comment by hepiyellow** [comment]
@garynewby 
Hi, do you have time to see if color-per-key is easy to implement?

**Comment by hepiyellow** [comment]
Another point for the new API suggestion, is that in order to reset all the highlighted keys, with the current API, I need to do:
```
keyboardView.highlightKeys([], color: HIGHLIGHT_COLOR, play: false)
```
Which forces me to pass a `color` and `play` which are irrelevant.

With the suggested api, it would be:
```
highlightKeys([])
```

**Comment by hepiyellow** [comment]
BTW: I found that setting the `highlightKeys` within `viewWillAppear` has no effect.
So I do it in `viewDidAppear`.
Is that proper?

**Comment by hepiyellow** [comment]
Hi,

I can make a PR for this.
Do you allow opening branches in your repo? Or should I fork it?

**Comment by garynewby** [comment]
The features addressed in 1.0.18
